src/utils/typeUtils.ts

```diff
@@ -9,3 +9,30 @@ export type RecursivePartial<T> = {
     T[P] extends object ? RecursivePartial<T[P]> :
       T[P];
 }
+
```

Comment:
I find this is a complex way[*] to do something like this:
```
if (/[^a-zA-Z]/.test(s)) throw new Error(...);
return lodash.snakeCase(key).toUpperCase();
```

* - meaning I don't understand what it's doing and think that's the wrong approach for a string-conversion function.

Reply:
Unfortunately, this does the conversion at runtime; we need to do it at compile-time (this is roughly the equivalent of C++ template metaprogramming).

Reply:
I didn't realize that. And the code immediately reminded me of C++ (not mentioning any guilty frameworks here).

If it works, I'm ok with it. But it seems like if the change is done at compile time we can't easily test it.

Reply:
Yeah, it seems to work. I test things by seeing that tooltips I get for `commit("thing…"` in VSCode, essentially.  That, and changing the code to be incorrect (like `commit('SET_SUDO_ALLOWED', 3)`) and checking that compilation fails (i.e. it catches the errors we want it to catch — here because `3` is not a boolean).